frostsnap_core/tests/malicious.rs

```diff
@@ -34,3 +36,69 @@ fn keygen_maliciously_replace_public_poly() {
         Err(frostsnap_core::Error::InvalidMessage { .. })
     ))
 }
+
+/// Send the same signing request to a device twice, asking for nonce reuse.
+/// The device should reject signing the second request.
+#[test]
+fn nonce_reuse() {
+    let threshold = 1;
+    let mut coordinator = FrostCoordinator::new();
+    let mut test_rng = ChaCha20Rng::from_seed([42u8; 32]);
+
+    let mut device = FrostSigner::new_random(&mut test_rng);
+    let device_ids = BTreeSet::from_iter([device.device_id()]);
+
+    let do_keygen_message = coordinator.do_keygen(&device_ids, threshold).unwrap();
+    let do_keygen_response = device.recv_coordinator_message(do_keygen_message).unwrap();
+
+    for message in do_keygen_response {
+        if let DeviceSend::ToCoordinator(message) = message {
+            let coordinator_responses = coordinator.recv_device_message(message).unwrap();
+
+            for response in coordinator_responses {
+                if let CoordinatorSend::ToDevice(message) = response {
+                    device.recv_coordinator_message(message).unwrap();
+                }
+            }
+        }
+    }
+
+    coordinator.keygen_ack(true).unwrap();
+    device.keygen_ack(true).unwrap();
+
+    let (_coordinator_sends, sign_request) = coordinator
+        .start_sign(SignTask::Plain(b"utxo.club!".to_vec()), device_ids.clone())
+        .unwrap();
+
+    let _device_responses = device
+        .recv_coordinator_message(sign_request.clone())
+        .unwrap();
+
+    let _device_sends = device.sign_ack(true).unwrap();
+
+    // Receive the same sign request again with the same nonces
```

Comment:
```suggestion
    // Receive the same sign request again with the same nonces
```
This comment is outdated :)